agent/structs/config_entry_discoverychain.go

```diff
@@ -871,6 +871,36 @@ type ServiceResolverConfigEntry struct {
 	RaftIndex
 }
 
+func (e *ServiceResolverConfigEntry) RelatedPeers() []string {
+	peers := make(map[string]struct{})
+
+	if r := e.Redirect; r != nil && r.Peer != "" {
+		peers[r.Peer] = struct{}{}
+	}
+
+	if e.Failover != nil {
+		for _, f := range e.Failover {
+			for _, t := range f.Targets {
+				if t.Peer != "" {
+					peers[t.Peer] = struct{}{}
+				}
+			}
+		}
+	}
+
+	if len(peers) == 0 {
+		return nil
+	}
+
+	out := make([]string, 0, len(peers))
+	for svc := range peers {
+		out = append(out, svc)
+	}
+	sort.Strings(out)
```

Comment:
Nit. Is there a reason for sorting? I feel like disco chains are already pretty heavy to compile. Also, I think we have a `maps.SliceOfKeys()` helper function to make this a little cleaner.

Reply:
Or we could just have this return a map